### PR TITLE
fix(DB) ICC Intro Event - Saurfang/Muradin MODELID

### PR DESCRIPTION
**Pull request by Akimboss**
Hello this is the fixe for #3261 and #3422 
The #3261 caused the bug #3422 it wasn't the right way to do it, and i call it even when they merged it first time, but no one listen till someone else reported the bug, so here's the fixe and good luck everyone

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1599321019527522400');
+UPDATE creature_template SET modelid1='30583' WHERE (entry='32315');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1599321019527522400.sql:2]: No.
Update the modelID in `creature` not `creature_template`.

Comment (Akimboss) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql]: I updated it

Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql]: please don't use strings where you can just use numbers:

```sql
UPDATE creature_template SET modelid1 = 30583 WHERE entry = 32315;
```

**Comment by FrancescoBorzi** [comment]
@Akimboss first of all many thanks for your contribution!

Just a suggestion for your future PRs, do not use `master` branch. Create 1 new branch per every PR.

You can follow this ;) https://www.azerothcore.org/wiki/How-to-create-a-PR

**Comment by Remberance** [comment]
tested, worked.

**Review by FrancescoBorzi — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql (hunk 1 above)
Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql]: @Akimboss please resolve @Kitzunu 's comment

**Comment by locus313** [comment]
tested. looks good.

**Comment by Goatform** [comment]
@FrancescoBorzi  @Akimboss 

So this is not fully correct but I think it won't matter in the long run.

NPC entry 32315 should keep the same model.
Wowhead source (picture): https://www.wowhead.com/npc=32315/high-overlord-saurfang

The NPC entry that was used in that change is correct, however it should not be used the way it was used (when you merged). You never remove SQL row just like that as you did in commit like this: 3a56cdbd4a

Source for entry 37187 - https://www.wowhead.com/npc=37187/high-overlord-saurfang
(You can even look at picture examples and you will see that it will show that NPC inside ICC)

So to sum this up, the problem with commit 3a56cdbd4a was:

1. SQL code itself... you are not supposed to remove 1 row ... changing manually entry ids like that) but instead, create a totally new row if the one does not exist in DB already.
2. When you are doing change like that you have to also update the C++ part (script) which he did not and somehow that commit was approved.

(The list of the problem is not referring to this PR but to the commit that caused the issue with all of this)

**Comment by peti446** [comment]
@Goatform is correct, the script needs to be updated to reflect the new ID or just rever the old GUID to use the specific ID and model, also the creature_template for 32315 does not appear to be in my database ?

**Comment by Remberance** [comment]
37187 maybe in the front of Saurfang(ICC 4TH boss)

32315 in the gate of icc? i guess.

**Comment by Akimboss** [comment]
> 37187 maybe in the front of Saurfang(ICC 4TH boss)
> 
> 32315 in the gate of icc? i guess.

Yuup, this query has been tested successfully, waiting to be merged 👍

**Comment by frytik** [comment]
If someone want to take look into full intro, you can here https://pastebin.com/raw/2rPZBrnu https://pastebin.com/raw/CaCvd3qj

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql (hunk 1 above)
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql]: I still stand by my comment here.

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1599321019527522400');
+
+UPDATE creature_template SET modelid1 = 30583 WHERE entry = 32315;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1599321019527522400.sql:3]: add \` around keywords

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql (hunk 2 above)
Comment (Akimboss) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql]: done

**Comment by FrancescoBorzi** [comment]
It's not clear to me whether @Goatform 's concerns have been addressed or there is still some disagreement towards this change (I haven't checked myself - currently I can't run the client).

**Comment by Kitzunu** [comment]
I still stand by reverting the first commit and then updating modelid in `creature`

**Comment by Remberance** [comment]
我还是说中文吧，我英文水平too  bad
原来那个32315是在大门前的萨鲁法尔大王，只是模型有问题。然而37187是在小萨鲁法尔前的大王，他会有开战的gossip，所以会影响非官方。

而且大门前出发弗丁喊话的npc entry不对自然会影响剧情触发。所以修改32175的模型(即本帖)是正解。

**Comment by peti446** [comment]
@Remberance I Used a translater to translate it so not sure if that will work, I tested this and on my end it did not fix it as the entry was still 37187, I am here with @Kitzunu and simply change the model ID and the npc entry back to 32315 as this is how the script is done:
`UPDATE 'creature' SET 'id' = '32315', 'modelid' = '30583' WHERE ('guid' = '200984');` I have tested this and it works like a charm

Also, I would like to mention that the given creature mentioned in this PR does not exits, so I assume we would also need to add to this PR the creature in the correct position after we change the template.

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1599321019527522400');
+
+UPDATE `creature_template` SET `modelid1` = 30583 WHERE `entry` = 32315;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1599321019527522400.sql:3]:
```suggestion
UPDATE `creature` SET `id` = 32315, `modelid` = 30583 WHERE `guid` = 200984;
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1599321019527522400.sql (hunk 3 above)
Comment (Goatform) [reply data/sql/updates/pending_db_world/rev_1599321019527522400.sql:3]: @Kitzunu 
``` sql
UPDATE `creature` SET `id` = 32315, `modelid` = 30583, `curhealth`='4183500' WHERE `guid` = 200984;
```
Do it like this in case some guys were going back and forth with fixing the issue temporarily.

**Comment by peti446** [comment]
The fix works, he has no longer the ui interaction and killing the two The Dammed  it triggers the interaction

**Comment by FrancescoBorzi** [comment]
so should this be merged?

**Comment by peti446** [comment]
@FrancescoBorzi I would say yes it fixes the problem and reverts it to what is used to be.

**Review by FrancescoBorzi — APPROVED** [review]
